src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1111,12 +1111,24 @@ public function newPivot(self $parent, array $attributes, $table, $exists, $usin
     /**
      * Determine if the model has a given scope.
      *
-     * @param  string  $method
+     * @param  string  $scope
      * @return bool
      */
-    public function hasScope($method)
+    public function hasScope($scope)
+    {
+        return method_exists($this, 'scope'.ucfirst($scope));
+    }
+
+    /**
+     * Apply the given named scope if possible.
+     *
+     * @param  string  $scope
+     * @param  array  $parameters
+     * @return mixed
+     */
+    public function callNamedScope($scope, array $parameters = [])
     {
-        return method_exists($this, 'scope'.ucfirst($method));
+        return $this->{'scope'.ucfirst($scope)}(...$parameters);
```

Comment:
This PR is WIP because I need to check this syntax is good. If anyone knows before I try it out, let me know.